float8_experimental/float8_dynamic_linear.py

```diff
@@ -30,63 +34,43 @@ def forward(
 
     @staticmethod
     def backward(ctx, gradY):
+        if tensor_already_casted_to_fp8(gradY):
```

Comment:
for `cast_to_float8_e5m2_bw`, unfortunately I can do a forward check only and have to check the backward gradients to see if it's already casted, as the forward only have `y` but not `grad_y`